### DISCUSSION TITLE
SeqrValidationError while uploading WES dataset

### DISCUSSION THREAD

**Discussion by anjalijain22** [broadinstitute/seqr#2966] [Q&A] [answered]
Hello,

I'm trying to upload a WES dataset to a local instance of seqr but while running the pipeline runner I get the following error -

`SeqrValidationError: Sample type validation error: dataset sample-type is specified as WES but appears to be WES because it contains many common non-coding variants`

To troubleshoot this, I tried uploading the dataset as a WGS and the pipeline ran successfully. 
However, I'm certain my dataset is whole exome, so I'm not sure why this error is coming up and if there was a way to switch off the validation step (is this recommended?)?

I'd appreciate your help with this!

Thanks,
Anjali

**Comment by hanars** [comment, marked as answer]
So there is a typo in the error message, it should say `dataset sample-type is specified as WES but appears to be WGS`. The actual check itself does appear to be working - this error is raised because the VCF you loaded has a "significant" number of both coding and noncoding variants, which we would generally not expect in WES.

That said, if you are confident that this WES data you can turn off validation and load it as such. You need to modify the executed command so it has the `--dont-validate` argument. Our metric for determining whether there are "significant" noncoding variants does depend on the total number of variants in a VCF and is optimized for large VCFs, so it is possible that if you are loading a relatively small VCF and therefore our validation is not valid

**Reply by pamelaxu213** [reply]
Hi Hana,

I am wondering where and in which script can I add the `--dont-validate` argument? Adding to `data_load.sh` did not work.

Thanks,
Pam

**Reply by hanars** [reply]
You have to manually construct the command to run, unfortunately the script does not accept the argument. See this discussion for an example: https://github.com/broadinstitute/seqr/discussions/3207#discussioncomment-4924417

**Comment by anjalijain22** [comment]
Hi @hanars! Is there a way to disable validation for a dataset in the new clickhouse backend? The pipeline says my WES dataset is WGS but I am certain its a WES dataset. Thanks in advance for your help!

**Reply by hanars** [reply]
@bpblanken this is not supported in the UI, can you explain the procedure for re-triggering a failed job with updated parameters?

**Reply by bpblanken** [reply]
Yes,  the cleanest way to skip validations is to:

1) Port forward to the pipeline runner api:
```
 kubectl port-forward services/pipeline-runner-api 6000:6000
```

2) Queue a request with `curl` that skips sample type validation:

```
curl -X POST 'http://localhost:6000/loading_pipeline_enqueue' -H "Content-Type: application/json" \
     -d '{
    "projects_to_run": [
        "your project guid"
    ],
    "dataset_type": "SNV_INDEL",
    "reference_genome": "GRCh38",
    "callset_path": "your callset",
    "sample_type": "WES",
     "validations_to_skip": ["validate_sample_type"]
}'
```

3) Your HTTP response should look like:
```
{"Successfully queued": {"request_type": "LoadingPipelineRequest", "attempt_id": 0, "callset_path": "your callset", "project_guids": ["your project guid"], "sample_type": "WES", "reference_genome": "GRCh38", "dataset_type": "SNV_INDEL", "skip_check_sex_and_relatedness": true, "skip_expect_tdr_metrics": true, "validations_to_skip": ["validate_sample_type"]}
```
